### ISSUE TITLE
RFC: Entity: provide changes with map function

### ISSUE THREAD

**Issue by aigosh** [ngrx/platform#671]
## I'm submitting a...

<pre><code>
[ ] Regression (a behavior that used to work and stopped working in a new release)
[ ] Bug report  
[x] Feature request
[ ] Documentation issue or request
</code></pre>

## What is the current behavior?

Current behaviour does not let to update entities depending on current entity state.

## Expected behavior:

So I propose to provide changes as map function like this:

```typescript
adapter.updateOne({
    id: someId,
    changes: entity => ({
         someValue: entity.someField + 1
    })
})

adapter.updateMany([
    {
        id: someId,
        changes: entity => ({
             someValue: entity.someField + 1
        })
    }
])
```

So the changes function for: 
```typescript
 EntityAdapter<T> 
``` 
will have the following signature :
```typescript
T => Partial<T>
```

**Comment by brandonroberts** [comment]
This is straightforward enough without adding more to the API. And `changes` is already a `Partial<T>` without the function.

**Comment by tdeschryver** [comment]
Is this still needed?
Personally I think this can easily be implemented with #907 and this change will be 'bloating' the API.
If many users need this I'm all in tho, I just think this won't be used by a lot of users.

**Comment by timdeschryver** [comment]
Closing this because https://github.com/ngrx/platform/pull/907 landed.
